pkg/receive/handler.go

```diff
@@ -682,6 +692,22 @@ func (h *Handler) RemoteWrite(ctx context.Context, r *storepb.WriteRequest) (*st
 	}
 }
 
+// relabel relabels the time series labels in the remote write request.
+func (h *Handler) relabel(wreq *prompb.WriteRequest) {
+	if len(h.options.RelabelConfigs) > 0 {
+		timeSeries := make([]prompb.TimeSeries, 0, len(wreq.Timeseries))
+		for _, ts := range wreq.Timeseries {
+			lbls := relabel.Process(labelpb.ZLabelsToPromLabels(ts.Labels), h.options.RelabelConfigs...)
+			if lbls == nil {
```

Comment:
If this is `nil` then we will have an empty `Timeseries` in the result? :thinking: don't we need to remove such elements from `timeSeries`?